### PR TITLE
bugfix/17139-chart-proj4

### PR DESCRIPTION
**Pull request by raf18seb**
Fixed https://github.com/highcharts/highcharts/issues/17139, `chart.proj4` option didn't work.

### CODE REVIEW

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Thanks for the PR! 

For legacy, proj4-based projection we already have tests in `/unit-tests/maps/geojson`. So it would be better if we adapt that to this case. It runs a roundtrip projection of five points, so I think the only adaptation we need to do is to run it with `proj4` as a chart setting in addition to the global property.

**Review by madepiet — APPROVED** [review]

File: samples/unit-tests/maps/geojson/demo.js [hunk 1]
```diff
@@ -1,8 +1,13 @@
 QUnit.test('GeoJSON and map transforms', assert => {
 
+    // #17139
+    const proj4 = window.proj4;
+    window.proj4 = undefined;
```
Comment (TorsteinHonsi) [inline samples/unit-tests/maps/geojson/demo.js:5]: So now we have a test for the `chart.proj4` option, but we no longer have a test for `window.proj4`. We need to refactor this so both cases are covered. Move the logic to a function, then run the same set of tests. Can probably run `chart.update` in between to change the setting.

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
We need to test both ways of defining 'proj4'.

File: samples/unit-tests/maps/geojson/demo.js (hunk 1 above)
Comment (raf18seb) [reply samples/unit-tests/maps/geojson/demo.js:5]: Sure! In progress

**Review by DJTechnoo — APPROVED** [review]

**Review by madepiet — APPROVED** [review]

**Review by KacperMadej — APPROVED** [review]

**Review by TorsteinHonsi — APPROVED** [review]

**Pull request merged**